storage/remote/client.go

```diff
@@ -257,22 +262,38 @@ func (c *client) Read(ctx context.Context, query *prompb.Query) (*prompb.QueryRe
 	if err != nil {
 		return nil, errors.Wrap(err, "error sending request")
 	}
+
+	contentType := httpResp.Header.Get("Content-Type")
+	if strings.HasPrefix(contentType, "application/x-protobuf") {
+		return c.handleSampledResponse(httpResp, start, req, sortSeries)
+	}
+
+	if httpResp.StatusCode/100 != 2 {
+		defer func() {
```

Comment:
this does not help.

Please look on this function and think about what are the cases when we need to close the body and exhaust? I think you are missing one.

I think because we run functions it's totally ok to defer this in all cases JUST after we get response. and let's make sure nothing closes twice inside functions WDYT?

Reply:
Thanks for taking a look at my code Bartek.
I think you are referring to closing the `httpRest.Body` over here:
```
func (c *client) handleStreamedResponse(httpResp *http.Response) (storage.SeriesSet, error) {
	stream := NewChunkedReader(httpResp.Body, DefaultChunkedReadLimit, nil)
	return &chunkedResponseSeriesSet{stream: stream}, nil
}
``` 
So I have to `close()` the `bufio.Reader` in `ChunkedReader`?

Reply:
PLTA! :-)

Reply:
Suggested something here: https://github.com/prometheus/prometheus/pull/7354#discussion_r494860692